FV3/atmos_cubed_sphere/tools/coarse_grained_restart_files.F90

```diff
@@ -147,11 +157,28 @@ subroutine register_fv_core_coarse(tile_count, hydrostatic, hybrid_z, &
 
     filename = 'fv_core_coarse.res.nc'
 
-    id_restart = register_restart_field(restart%fv_core_coarse, &
+    !--- optionally include D-grid winds even if restarting from A-grid winds
+    if (optional_dgrid_vel_rst .and. restart_from_agrid_winds) then
+       id_restart = register_restart_field(restart%fv_core_coarse, &
+            filename, 'u', restart%u, domain=coarse_domain, position=NORTH, &
+            mandatory=.false., tile_count=tile_count)
```

Comment:
How is this `mandatory` flag used? It's a bit confusing, because what is "mandatory" depends on the configuration of the run using these restart files, not the configuration of the run that is saving them.

Reply:
Also, could have a bit less repeated code if we first determine which velocity restarts are "mandatory" and save that info to local booleans (i.e. `agrid_winds_mandatory` and `dgrid_winds_mandatory`) for use in `register_restart_field`.

Reply:
That's right -- the `mandatory` flag is used to indicate when a particular restart field is required to start the model in a given configuration.  E.g. if I configured the model to restart from A-grid winds, but the restart files didn't include them, it should crash, but nothing should happen if the files do not contain D-grid winds (the same goes in reverse if the model is configured to restart from D-grid winds, i.e. as it is by default).  Does that make sense?

Reply:
Agreed that the `mandatory` flag could be set using dynamically generated booleans.  Will update things accordingly.

Reply:
> Agreed that the mandatory flag could be set using dynamically generated booleans. Will update things accordingly.

Actually I take this back.  I want to maintain the ability to forgo registering the optional restart fields entirely.  If I register them as `mandatory = .false.` they will still be written out even though they are not required to restart the model.  If I do not register them they will not be written out (this saves space if desired).

This does make me pause a bit though to think about how we might use these flags in practice.  I think we may want to decouple the fine restart files from the coarse ones in this context.  Thinking ahead to a situation like the one-year high-res simulation, we may want to run that in segments, where we restart the full-resolution model from D-grid winds, but write out coarse-resolution files with only A-grid winds (if we decide that we want to go with that strategy and want to save space).  The way things are structured currently, we'd be bound to outputting both A-and-D grid winds on all restart files.

Reply:
Added separate namelist flags to control what gets written to coarse restart files:
- `write_coarse_dgrid_vel_rst` (default true)
- `write_coarse_agrid_vel_rst` (default false)